### ISSUE TITLE
Doesn't work under node@4.x

### ISSUE THREAD

**Issue by lazutkin** [bcherny/json-schema-to-typescript#73]
Latest release use some features that unsupported by node@4.x - ES6 default parameters for instance. So there is an error in attempt to run cli.js

```
.../.nvm/versions/node/v4.7.2/lib/node_modules/json-schema-to-typescript/dist/src/index.js:30
function compileFromFile(filename, options = exports.DEFAULT_OPTIONS) {
                                           ^

SyntaxError: Unexpected token =
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)

```

**Comment by bcherny** [comment]
Thanks for the bug report. There are a few issues:

1. Program is compiled to ES6, should be ES5
2. Ava only works on Node >= 4.5 (https://github.com/avajs/ava/issues/1354)
3. Object key iteration algorithm changed somewhere between Node 4 and 6, producing different output in some cases. Specifically, the realWorld.swagger test case fails
4. Manually reproducing the error in (3) triggers a circular dereferencing bug in json-schema-ref-parser (https://github.com/BigstickCarpet/json-schema-ref-parser/issues/37)

I will keep this issue updated with progress. For now, I would recommend generating Node 6 if you can (especially since Node 4 has [officially entered maintenance mode](https://github.com/nodejs/LTS#lts-schedule1)).

**Comment by bcherny** [comment]
Published as 4.1.2.
